portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,33 +22,44 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private static final Gson gson = new Gson();
+    private static final Gson GSON = new Gson();
+    private static final String COMMENT_INPUT = "comment-input";
     private ArrayList<String> messages = new ArrayList<String>();
+  
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String json = convertToJsonUsingGson(messages);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
 
     @Override
-    public void init() {
-        messages.add("message 1");
-        messages.add("message 2");
-        messages.add("According to all known laws of aviation, there is no way a bee should be able to fly. " + 
-                    "Its wings are too small to get its fat little body off the ground. The bee, of course, flies anyway " +
-                    "because bees don't care what humans think is impossible.");
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String text = getParameter(request, COMMENT_INPUT, " ");
+        messages.add(text);
+        response.sendRedirect("/index.html");
     }
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(messages);
-    response.setContentType("application/json;");
-    response.getWriter().println(json);
-  }
+    /**
+    * @return the request parameter
+    */
+    private String getParameter(HttpServletRequest request, String name) {
+        String value = request.getParameter(name);
+        if (value == null) {
+            return defaultValue;
```

Comment:
what is defaultValue?

Reply:
its not declared on the function

Reply:
It represents the response given if the user did not input a text. Would you prefer that I add that explanation to the javadoc or remove defaultValue entirely?

Reply:
I think this will not compile because defaultValue is not decleared here
```
private String getParameter(HttpServletRequest request, String name) {
```

Reply:
Oh that's what you meant! You're right, I completely forgot to put defaultValue back into the function. Thank you for pointing this out!